### PR TITLE
add Makefile

### PR DESCRIPTION
**Pull request by blackmorth**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+start: ## Starts the docker containers
+	docker-compose up -d
+
+stop: ## Stops the docker containers
+	docker-compose stop
+
+clean: ## Deletes the docker containers and clean the project
+	docker-compose down --remove-orphans
+
+uninstall: stop clean ## Uninstalls the project
```
Comment (Livda) [inline Makefile]: Une ligne blanche en fin de fichier please

File: symfony/Makefile [hunk 2]
```diff
@@ -0,0 +1,30 @@
+.DEFAULT_GOAL := help
+
+##
+## Variable
+##
+PHP = docker-compose exec php
```
Comment (Livda) [inline symfony/Makefile]: `docker-compose exec -u www-data php` pour conserver les permissions sur les fichiers

File: symfony/Makefile [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+vendor: composer.json composer.lock ## Install vendors
+	$(PHP) composer install
+
+composer.lock: composer.json
+	$(PHP) composer update
+
+.PHONY: cs
+cs: | vendor ## Runs the php-cs-fixer
+	$(PHP) php vendor/bin/php-cs-fixer fix --allow-risky=yes
```
Comment (Livda) [inline symfony/Makefile]: Idem, ligne blanche en fin de fichier

File: install.sh [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    echo Which port do you want to use for nginx?
+    read -p 'Nginx port: ' nginx
+    if [[ ! -z ${nginx} ]]
+    then
+        nginx_port=$(cut -d':' -f1 <<< ${services__nginx__ports__1}| cut -d'"' -f2 )
+        echo ${nginx_port}
+        sed -i "s/\"${nginx_port}:80\"/\"${nginx}:80\"/g" docker-compose.yml
+        echo ok we just changed it to ${nginx}
+    fi
+fi
```
Comment (Livda) [inline install.sh]: Ligne blanche ici !

File: php-fpm/Dockerfile [hunk 5]
```diff
@@ -1,6 +1,6 @@
 FROM alpine:3.8
 
-
+RUN apk add tzdata && cp /usr/share/zoneinfo/Europe/Paris /etc/localtime && echo "Europe/Paris" >  /etc/timezone && apk del tzdata
```
Comment (Livda) [inline php-fpm/Dockerfile]: Tu peux ajouter `tzdata` dans la liste en dessous, faire ton cp après et supprimer `tzdata` ensuite ? J'aime bien avoir qu'un seul appel pour récupérer toutes mes dépendances !

File: install.sh [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        echo "127.0.0.1       ${domain}" | sudo tee --append /etc/hosts
+        echo ok we just added ${domain} to /etc/hosts
+    else
+        echo "127.0.0.1       www.changeme.local www.changeme.com" | sudo tee --append /etc/hosts
+        echo ok we just added www.changeme.local www.changeme.com to /etc/hosts
+    fi
+fi
+
+#Ask the user for the port
+read -p "Do you want to change default ports? [Y/N]" -n 1 -r
```
Comment (Livda) [inline install.sh]: Pas besoin de cette partie. Si l'utilisateur veut modifier les ports ou ajouter un reverse proxy, il est grand, il modifie directement le docker-compose.yml avec ses petits doigts. Pas besoin de jouer à la nounou :stuck_out_tongue:

File: install.sh [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+read -p "Do you want to add domain name to your /etc/hosts? [Y/N]" -n 1 -r
+echo
+if [[ $REPLY =~ ^[Yy]$ ]]
+then
+    echo which domain do you want to use for the project?
+    read -p 'domain name separed by space[www.changeme.local www.changeme.com]: ' domain
+    if [[ ! -z ${domain} ]]
+    then
+        echo "127.0.0.1       ${domain}" | sudo tee --append /etc/hosts
```
Comment (Livda) [inline install.sh]: Si l'utilisateur n'est pas sous Linux, ça ne fonctionne pas ça. Typiquement sous Mac, comme tu fais tourner docker dans une VM, il faut lui filer l'ip de la carte réseau associée à la VM. Il a moyen de vérifier le système de l'utilisateur avant de modifier ça ?

File: docker-compose.yml [hunk 8]
```diff
@@ -24,8 +24,8 @@ services:
         networks:
             - default
         volumes:
-            - ./symfony:/var/www/symfony
-            - ./logs/symfony:/var/www/symfony/var/log
+            - ./symfony:/var/www
+            - ./logs/symfony:/var/www/var/log
```
Comment (Livda) [inline docker-compose.yml:29]: Bien vu pour ça !

File: install.sh [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+        -e ":1;s|^\($s\)\($w\)$s:$s\[$s\(.*\)$s,$s\(.*\)$s\]|\1\2: [\3]\n\1  - \4|;t1" \
+        -e "s|^\($s\)\($w\)$s:$s\[$s\(.*\)$s\]|\1\2:\n\1  - \3|;p" $1 | \
+   sed -ne "s|,$s}$s\$|}|" \
+        -e ":1;s|^\($s\)-$s{$s\(.*\)$s,$s\($w\)$s:$s\(.*\)$s}|\1- {\2}\n\1  \3: \4|;t1" \
+        -e    "s|^\($s\)-$s{$s\(.*\)$s}|\1-\n\1  \2|;p" | \
+   sed -ne "s|^\($s\):|\1|" \
+        -e "s|^\($s\)-$s[\"']\(.*\)[\"']$s\$|\1$fs$fs\2|p" \
+        -e "s|^\($s\)-$s\(.*\)$s\$|\1$fs$fs\2|p" \
+        -e "s|^\($s\)\($w\)$s:$s[\"']\(.*\)[\"']$s\$|\1$fs\2$fs\3|p" \
+        -e "s|^\($s\)\($w\)$s:$s\(.*\)$s\$|\1$fs\2$fs\3|p" | \
```
Comment (Livda) [inline install.sh]: On va dire que j'ai compris l'expression régulière au-dessus :wink:

**Review by Livda — CHANGES_REQUESTED** [review]
Some improvements to do !

File: php-fpm/Dockerfile [hunk 10]
```diff
@@ -28,23 +27,42 @@ RUN apk add --update \
...
     libsodium \
     make \
-    curl
+    curl \
+    shadow \
+    tzdata \
+    sudo \
+    zsh \
+	git \
+	bash
```
Comment (Livda) [inline php-fpm/Dockerfile]: I've chosen alpine to get a minimal image. But install `bash`, `zsh`, `oh-my-zsh` and all that's stuff is not mandatory. If you want to interact with your container, use `sh`. It's not as sexy as zsh with a cool theme, but I don't need to run that on my production server.

File: php-fpm/Dockerfile [hunk 11]
```diff
@@ -28,23 +27,42 @@ RUN apk add --update \
...
 
+RUN cp /usr/share/zoneinfo/Europe/Paris /etc/localtime && echo "Europe/Paris" >  /etc/timezone && apk del tzdata
 RUN rm -rf /var/cache/apk/* && rm -rf /tmp/*
 
-ADD symfony.ini /etc/php7/php-fpm.d/
-ADD symfony.ini /etc/php7/cli/conf.d/
+RUN addgroup -g 1000 www-data && adduser -u 1000 -G www-data -D www-data -s zsh
+RUN groupmod -g 1000 www-data 
+RUN echo "%sudo	ALL=(ALL:ALL) ALL" >> /etc/sudoers
+RUN usermod -u 1000 -g www-data www-data && echo "www-data:www-data" | chpasswd && addgroup sudo && adduser www-data sudo
```
Comment (Livda) [inline php-fpm/Dockerfile]: Nope nope nope ! I don't want to run my application with root privileges. `www-data` needs to stay unprivileged. If we needs to run something which needs root privileges, just use `docker-compose exec php` to run it. This application should never have to manipulates things that require root privileges.

File: php-fpm/Dockerfile [hunk 12]
```diff
@@ -28,23 +27,42 @@ RUN apk add --update \
...
+    zsh \
+	git \
+	bash
 
+RUN cp /usr/share/zoneinfo/Europe/Paris /etc/localtime && echo "Europe/Paris" >  /etc/timezone && apk del tzdata
 RUN rm -rf /var/cache/apk/* && rm -rf /tmp/*
 
-ADD symfony.ini /etc/php7/php-fpm.d/
-ADD symfony.ini /etc/php7/cli/conf.d/
+RUN addgroup -g 1000 www-data && adduser -u 1000 -G www-data -D www-data -s zsh
```
Comment (Livda) [inline php-fpm/Dockerfile]: There is no needs to an `home` for our `www-data` user

File: php-fpm/Dockerfile [hunk 13]
```diff
@@ -28,23 +27,42 @@ RUN apk add --update \
...
+RUN groupmod -g 1000 www-data
```
Comment (Livda) [inline php-fpm/Dockerfile]: Why create the `www-data` group a second time here ? It's already created line 43

File: php-fpm/Dockerfile [hunk 14]
```diff
@@ -28,23 +27,42 @@ RUN apk add --update \
...
+ADD files/symfony.pool.conf /etc/php7/php-fpm.d/
 
-CMD ["php-fpm7", "-F"]
+RUN rm -f /etc/php7/php-fpm.d/www.conf
+RUN chown -R www-data:www-data /home/www-data
 
-WORKDIR /var/www/symfony
-EXPOSE 9001
+ADD files/php-fpm.sh /usr/bin/
+ENTRYPOINT ["/usr/bin/php-fpm.sh"]
```
Comment (Livda) [inline php-fpm/Dockerfile]: Leave `php-fpm` manager run as root. There is no needs to run it in user space and it's avoid some odd bugs. Our application doesn't run with privileges due to the configuration in `symfony.pool.conf`.

File: docker-compose.yml [hunk 15]
```diff
@@ -1,42 +1,51 @@
...
+        environment:
+            - COMPOSER_MEMORY_LIMIT=-1
     nginx:
         build: ./nginx
-        container_name: nginx
```
Comment (Livda) [inline docker-compose.yml:29]: why remove all the container_name ? It's pretty usefull if you wants to manipulate your containers outside of the project folder. And `docker-symfony_nginx_1` is pretty long to write !

File: php-fpm/Dockerfile [hunk 16]
```diff
@@ -28,23 +27,42 @@ RUN apk add --update \
...
+RUN echo "%sudo	ALL=(ALL:ALL) ALL" >> /etc/sudoers
+RUN usermod -u 1000 -g www-data www-data && echo "www-data:www-data" | chpasswd && addgroup sudo && adduser www-data sudo 
+
+RUN mkdir /run/php && chown -R www-data:www-data /run/php
 
-RUN addgroup www-data
-RUN adduser -u 1000 -D -g '' -G www-data www-data
+RUN chsh -s /bin/zsh www-data
+RUN sudo -u www-data sh -c "$(curl -fsSL https://raw.github.com/robbyrussell/oh-my-zsh/master/tools/install.sh)"
+RUN curl -sS https://getcomposer.org/installer | php7
```
Comment (Livda) [inline php-fpm/Dockerfile]: Just use the composer package from alpine. There is no really needs to re-downloaded after.

File: php-fpm/Dockerfile [hunk 17]
```diff
@@ -28,23 +27,42 @@ RUN apk add --update \
...
+RUN mv composer.phar /usr/bin/composer
+RUN chmod +x /usr/bin/composer && composer global require hirak/prestissimo
```
Comment (Livda) [inline php-fpm/Dockerfile]: Cool feature, but not needed here ! We put only the bare minimum in this image

File: php-fpm/files/symfony.ini [hunk 18]
```diff
@@ -0,0 +1,11 @@
+date.timezone = Europe/Paris
+xdebug.max_nesting_level = 1000
+xdebug.remote_enable = 1
+xdebug.remote_connect_back = 1
+xdebug.profiler_output_dir = "/var/logs/xdebug"
+xdebug.idekey = PHPSTORM
```
Comment (Livda) [inline php-fpm/files/symfony.ini]: There is no xdebug in this image, what's the purpose of this conf ?

File: php-fpm/files/symfony.ini [hunk 19]
```diff
@@ -0,0 +1,11 @@
...
+memory_limit = 512M
+display_errors = on
+realpath_cache_size=4096k
+realpath_cache_ttl=600
+opcache.max_accelerated_files=20000
```
Comment (Livda) [inline php-fpm/files/symfony.ini]: Blank line at the end ! Use the editorconfig extension for your editor, there is a .editorconfig are the root of the project which take care for you to this :wink:

File: php-fpm/files/symfony.pool.conf [hunk 20]
```diff
@@ -17,8 +17,7 @@ group = www-data
 ;                            specific port;
 ;   '/path/to/unix/socket' - to listen on a unix socket.
 ; Note: This value is mandatory.
-listen = 9001
-
+listen = 0.0.0.0:9000
```
Comment (Livda) [inline php-fpm/files/symfony.pool.conf]: `listen = 9000` == `listen = 0.0.0.0:9000`. Why to precise it ?

File: symfony/Makefile [hunk 21]
```diff
@@ -0,0 +1,31 @@
+.DEFAULT_GOAL := help
+
+##
+## Variable
+##
+PHP = docker-compose exec php
```
Comment (Livda) [inline symfony/Makefile]: `docker-compose exec -u www-data php`

**Review by Livda — CHANGES_REQUESTED** [review]

File: php-fpm/Dockerfile (hunk 14 above)
Comment (blackmorth) [reply php-fpm/Dockerfile]: in fact, it's for mac user. but ok I'll fix it

File: php-fpm/Dockerfile (hunk 17 above)
Comment (blackmorth) [reply php-fpm/Dockerfile]: You don't want to speed up you composer update or install ? 
https://github.com/hirak/prestissimo#benchmark-example

File: php-fpm/Dockerfile [hunk 22]
```diff
@@ -28,23 +27,27 @@ RUN apk add --update \
     php7-tokenizer \
     php7-xml \
     php7-zlib \
-    composer \
     libsodium \
     make \
-    curl
+    curl \
+    tzdata \
+	git
```
Comment (Livda) [inline php-fpm/Dockerfile]: no needs to have git

**Review by Livda — CHANGES_REQUESTED** [review]

**Pull request closed without merging**